### PR TITLE
Print diff between expected and actual messages in e2e tests

### PR DESCRIPTION
**Pull request by wangyangjun**
## Description
The return type of function MessageAssertion is updated from `bool` to `(bool, int, string)`.
Other corresponding changes are made base on this change. 
- The returned `int` value indicates the number of matched blocks between expect message and actual got message.
- The returned `string` value is the diff between expect message and actual got message.

This PR compares number of matched blocks and print the diff corresponding to the maximum
value of number of matched blocks.

## Related issue
Resolves: #772

### CODE REVIEW

**Comment by wangyangjun** [comment]
One example of failed e2e test case result:

Screenshot 2022-10-12 at 23 21 34
Screenshot 2022-10-12 at 23 21 54

**Review by pkosiec — COMMENTED** [review]
Great Job Reaction - XreQmk7ETCak0

I'm still finalizing testing, but that looks really good! A few small comments.

File: test/e2e/k8s_helpers_test.go [hunk 1]
```diff
@@ -171,3 +172,34 @@ func updateEnv(existing []v1.EnvVar, env []v1.EnvVar, remove []string) []v1.EnvV
...
+// countMatchBlock count the number of lines matched between two strings
+func countMatchBlock(expect string, actual string) int {
+	matcher := difflib.NewMatcher(difflib.SplitLines(expect), difflib.SplitLines(actual))
+	matches := matcher.GetMatchingBlocks()
+	count := 0
+	for _, match := range matches {
+		count += match.Size
+	}
+	return count
+}
```
[the remark this answers is not in the document]
Comment (wangyangjun) [reply test/e2e/k8s_helpers_test.go]: Nice suggestion.

File: test/e2e/slack_driver_test.go [hunk 2]
```diff
@@ -135,26 +135,41 @@ func (s *slackTester) InviteBotToChannel(t *testing.T, channelID string) {
 }
```
[the remark this answers is not in the document]
Comment (wangyangjun) [reply test/e2e/slack_driver_test.go:135]: Do we support diff for msg only or support diff for title, color and msg?

[the remark this answers is not in the document]
Comment (wangyangjun) [reply test/e2e/slack_driver_test.go:135]: From this comment, it seems we only support diff for msg -- https://github.com/kubeshop/botkube/blob/7cfcfba0ac82bdb44c6451f6de54d0a2434ddcf3/test/e2e/slack_driver_test.go#L273

[the remark this answers is not in the document]
Comment (pkosiec) [reply test/e2e/slack_driver_test.go:135]: @wangyangjun Oh you're right. Skip the `WaitForMessagePostedWithFileUpload` then 🙂 
And yes, I was referring to the message. Thank you!

**Review by pkosiec — APPROVED** [review]
Tested once again and it works well: https://github.com/pkosiec/botkube/actions/runs/3247955891/jobs/5328621321

Thank you very much for your contribution @wangyangjun!

File: test/e2e/discord_driver_test.go [hunk 3]
```diff
@@ -257,6 +274,8 @@ func (d *discordTester) WaitForMessagePostedWithAttachment(userID, channelID str
 
 	var fetchedMessages []*discordgo.Message
 	var lastErr error
+	var diffMessage string
+	common := -1
```
Comment (pkosiec) [inline test/e2e/discord_driver_test.go:278]: Can we revert this change to the previous form? I think the behavior when initial `common` is 0 is better, as we won't print a diff when there are no common blocks. What do you think?

```suggestion
	var common int
```

File: test/e2e/slack_driver_test.go [hunk 4]
```diff
@@ -245,6 +265,9 @@ func (s *slackTester) WaitForMessagePostedWithFileUpload(userID, channelID strin
 func (s *slackTester) WaitForMessagePostedWithAttachment(userID, channelID string, assertFn AttachmentAssertion) error {
 	var fetchedMessages []slack.Message
 	var lastErr error
+	var diffMessage string
+	common := -1
```
Comment (pkosiec) [inline test/e2e/slack_driver_test.go:269]: Same here:
```suggestion
	var common int
```

**Review by pkosiec — COMMENTED** [review]
Hey @wangyangjun, just two quick comments as I missed the recent change to have `-1` as the initial common block count. Appreciate your patience!

I could apply the suggestion by myself and merge this, but I'd like to double check with you whether you spot some issues with that, as you changed it to -1?

File: test/e2e/discord_driver_test.go (hunk 3 above)
Comment (wangyangjun) [reply test/e2e/discord_driver_test.go:278]: @pkosiec actually the original code is what you suggested. The reason I changed this in my last commit is there is only one line in the message of `AttachmentAssertion`. If there is any difference, the common will be always be `0`, then no diff will be print out. What do you think?

Comment (wangyangjun) [reply test/e2e/discord_driver_test.go:278]: If we don't want print diff there is only one message, feel free to apply the suggestions and merge the PR.

Comment (pkosiec) [reply test/e2e/discord_driver_test.go:278]: Thank you @wangyangjun, very good point. I think I found a solution - check the message limits. See this commit and tell me if that looks OK for you 👍 
https://github.com/kubeshop/botkube/pull/811/commits/e5300cf02db8cb7040d7c364574fd488fa623786

Comment (wangyangjun) [reply test/e2e/discord_driver_test.go:278]: @pkosiec Yes, it looks good to me.

Comment (pkosiec) [reply test/e2e/discord_driver_test.go]: Cool, I've just pushed the same changes for Discord because I forgot to do it.. Enabling auto-merge then!
